x-pack/plugins/fleet/server/services/settings.ts

```diff
@@ -70,7 +71,8 @@ export async function settingsSetup(soClient: SavedObjectsClientContract) {
 
 export async function saveSettings(
   soClient: SavedObjectsClientContract,
-  newData: Partial<Omit<Settings, 'id'>>
+  newData: Partial<Omit<Settings, 'id'>>,
+  options?: SavedObjectsUpdateOptions<SettingsSOAttributes>
```

Comment:
Are we using this? I think we should do optimistic concurrency control using the `version` option.

Also, I think if we get a `404` for saved settings (they don't exist yet) we should not do a create call with `overwrite: true` when migrating. We can add an option `createWithOverwrite: boolean` that defaults to `true`, but we can set it to `false` for the migration case.

Reply:
If the user is a Fleet user the settings object should exists so that scenario is probably never going to happen, but I could add that option for safety